libqtile/layout/xmonad.py

```diff
@@ -858,6 +858,16 @@ def right(self):
         self.clients.current_client = self._get_closest(x, y, candidates)
         self.group.focus(self.clients.current_client)
 
+    @expose_command()
+    def up(self):
+        """Focus on the closest window above the current window"""
+        self.previous()
+
+    @expose_command()
+    def down(self):
+        """Focus on the closest window below the current window"""
+        self.next()
+
```

Comment:
You're probably correct, but why is this needed. The previous decorators should have had the same effect as this. Does it cause issues for `MonadWide` on inheritance?

Reply:
Exactly, the decorators caused issues with `MonadWide` on inheritance. I tried overriding in `MonadWide` and leaving the decorators as is, but the overrides were ignored and the decorators in `MonadTall` took precedence (confirmed by adding some log lines). Avoiding decorators allowed inheritance and overrides to work as usual.